src/markdown/generator.ml

```diff
@@ -0,0 +1,382 @@
+
+let entity e =
+  match e with "#45" -> escaped "-" | "gt" -> str ">" | s -> str "&%s;" s
+
+let raw_markup (_ : Raw_markup.t) = noop
+
+let style (style : style) content =
+  match style with
+  | `Bold -> string "**" ++ (content ++ str "**")
+  | `Italic | `Emphasis -> string "_" ++ (content ++ str "_")
+  | `Superscript -> string "<sup>" ++ content
+  | `Subscript -> string "<sub>" ++ content
+
+let make_hashes n = String.make n '#'
+
+type args = { generate_links : bool ref; md_flavour : string ref }
+
+let args = { generate_links = ref true; md_flavour = ref "" }
+
+let rec source_code (s : Source.t) nbsp =
+  match s with
+  | [] -> noop
+  | h :: t -> (
+      let continue s = if s = [] then concat [] else source_code s nbsp in
+      match h with
+      | Source.Elt i -> inline i nbsp ++ continue t
+      | Tag (None, s) -> continue s ++ continue t
+      | Tag (Some _, s) -> continue s ++ continue t)
+
+and inline (l : Inline.t) nbsp =
+  match l with
+  | [] -> noop
+  | i :: rest -> (
+      let continue i = if i = [] then noop else inline i nbsp in
+      let make_link c s =
+        open_sq_bracket ++ continue c ++ close_sq_bracket ++ open_parenthesis
+        ++ string s ++ close_parenthesis ++ continue rest
+      in
+      let cond then_clause else_clause =
+        if !(args.generate_links) then then_clause else else_clause
+      in
+      match i.desc with
+      | Text "" -> continue rest
+      | Text s -> (
+          match s with
+          | "end" | "}" | "]" ->
+              string (make_hashes 6) ++ space ++ nbsp ++ string s
+          | _ ->
+              let l, _, rest =
+                Doctree.Take.until l ~classify:(function
+                  | { Inline.desc = Text s; _ } -> Accum [ str "%s" s ]
+                  | _ -> Stop_and_keep)
+              in
+              concat l ++ continue rest)
+      | Entity e ->
+          let x = entity e in
+          x ++ continue rest
+      | Styled (sty, content) -> style sty (continue content) ++ continue rest
+      | Linebreak -> break ++ continue rest
+      | Link (href, content) ->
+          cond
+            (match content with
+            | [] -> noop
+            | i :: rest ->
+                (match i.desc with
+                | Text _ -> make_link content href
+                | _ -> continue content ++ continue rest)
+                ++ continue rest)
+            (continue content ++ continue rest)
+      | InternalLink (Resolved (link, content)) ->
+          cond
+            (match link.page.parent with
+            | Some _ -> continue content ++ continue rest
+            | None -> make_link content (make_hashes 1 ^ link.anchor))
+            (continue content ++ continue rest)
+      | InternalLink (Unresolved content) -> continue content ++ continue rest
+      | Source content ->
+          cond
+            (source_code content nbsp ++ continue rest)
+            (backticks ++ source_code content nbsp ++ backticks ++ continue rest)
```

Comment:
Why does this change formatting depending on generate-links ? I see no reason.

Reply:
That’s because we would end up with links in back ticks which will confuse the reader, that is, `` [text](href) `` (`text` will be a link, but it won’t be displayed as a link rather as source code.

Reply:
Do you have an example of links inside "source code" ?
Anyway, removing the backticks isn't the solution because we want them.